submarine-server/server-core/src/main/java/org/apache/submarine/server/environment/EnvironmentManager.java

```diff
@@ -0,0 +1,226 @@
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException(Status.BAD_REQUEST.getStatusCode(),
+          "Unable to process the environment spec.");
+    }
+  }
+  
+  /**
+   * Update environment
+   * @param name Name of the environment
+   * @param spec environment spec
+   * @return Environment environment
+   * @throws SubmarineRuntimeException the service error
+   */
+  public Environment updateEnvironment(String name, EnvironmentSpec spec)
+      throws SubmarineRuntimeException {
+    Environment env = getEnvironmentDetails(name);
+    if (env == null) {
+      throw new SubmarineRuntimeException(Status.NOT_FOUND.getStatusCode(),
+          "Environment not found.");
+    }
+    checkSpec(spec);
+    
+    EnvironmentEntity entity = new EnvironmentEntity();
+    entity.setEnvironmentName(spec.getName());
+    entity.setEnvironmentSpec(
+        new GsonBuilder().disableHtmlEscaping().create().toJson(spec));
+    try (SqlSession sqlSession = MyBatisUtil.getMetastoreSqlSession()) {
+      EnvironmentMapper environmentMapper =
+          sqlSession.getMapper(EnvironmentMapper.class);
+      int environmentId = environmentMapper.update(entity);
+      sqlSession.commit();
+
+      Environment updatedEnvironment = new Environment();
+      updatedEnvironment.setName(spec.getName());
+      updatedEnvironment.setEnvironmentId(environmentId);
+      updatedEnvironment.setEnvironmentSpec(spec);
+      return updatedEnvironment;
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException(Status.BAD_REQUEST.getStatusCode(),
+          "Unable to process the environment spec.");
+    }
+  }
+  
+  /**
+   * Delete environment
+   * @param name Name of the environment
+   * @return Environment environment
+   * @throws SubmarineRuntimeException the service error
+   */
+  public Environment deleteEnvironment(String name)
+      throws SubmarineRuntimeException {
+    Environment env = getEnvironmentDetails(name);
+    if (env == null) {
+      throw new SubmarineRuntimeException(Status.NOT_FOUND.getStatusCode(),
+          "Environment not found.");
+    }
+    
+    try (SqlSession sqlSession = MyBatisUtil.getMetastoreSqlSession()) {
+      EnvironmentMapper environmentMapper =
+          sqlSession.getMapper(EnvironmentMapper.class);
+      environmentMapper.delete(name);
+      sqlSession.commit();
+      return env;
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException(Status.BAD_REQUEST.getStatusCode(),
+          "Unable to delete the environment.");
+    }
+  }
+  
+  /**
+   * Get Environment
+   * @param name Name of the environment
+   * @return Environment environment
+   * @throws SubmarineRuntimeException the service error
+   */
+  public Environment getEnvironment(String name)
+      throws SubmarineRuntimeException {
+    Environment environment = getEnvironmentDetails(name);
```

Comment:
Is this always load from DB, is it better to maintain an in-mem cache?

Reply:
Taken care.